nextjs-13/src/app/page.tsx

```diff
@@ -1,20 +1,36 @@
 'use client';
-import { useState } from 'react';
+import { Suspense, useEffect, useState } from 'react';
+import { ClipLoader } from 'react-spinners';
 
 import Head from 'next/head';
 import { redirect } from 'next/navigation';
 import { useSession } from 'next-auth/react';
 
 import CreateNewsletterModal from '@components/CreateNewsletterModal';
+import ListNewsletter from '@components/ListNewsletter';
+import requestManager from 'lib/request/manager';
+import promiseWrapper from 'lib/request/promiseWrapper';
 
 const Home = () => {
   const { status } = useSession();
   const [modalIsOpen, setIsOpen] = useState(false);
+  const [promise, setPromise] = useState();
 
-  if (status === 'loading') {
-    return <p>Loading...</p>;
+  async function fetchRecords() {
+    const res = await requestManager('GET', 'v1/newsletter');
+
+    return res.records;
+  }
+
+  async function getData() {
+    setPromise(promiseWrapper(fetchRecords()));
```

Comment:
I do not understand why this `promiseWrapper` is required?

Reply:
Ya I should have put the explanation for this on the description.

The `promiseWrapper` is exclusively to make the `Suspense` work.

When the component inside `Suspense` is fetching data(`ListNewsletter`), we need to throw a pending `Promise` so that `Suspense` can catch and can trigger the loading. That's the purpose of the `promiseWrapper` also.

And as I tested, it's true that without the `promiseWrapper`, the `Suspense` is never triggered and the `Loading` never appears.

As I research about `Suspense`, I mainly use this 2 blog posts:
- [What Is React Suspense](https://prismic.io/blog/what-is-react-suspense)
- [React Suspense for Data Fetching with Axios](https://deadsimplechat.com/blog/react-suspense/)

I also read other posts, and all these blog posts say kind of the same thing too!

As for the the reason I pass the `promise` instead of `records` in [this comment](https://github.com/nimblehq/js-framework-benchmark/pull/110#discussion_r1227485608), it's because if I instead pass the `records` to `ListNewsletter`, the `Suspense` never triggers.

Maybe it's because for the `Suspense` to work, the `Promise` need to be thrown INSIDE the component that `Suspense` is wrapping, aka `ListNewsletter`.

I kind of find using `Suspense` a little more troublesome than the normal way(use a `loading` state instead), but indeed it makes our code more concise.

What do you think about this? 👀

Reply:
This info helps a lot. It does look more complicated, but I also find that the implementation in https://deadsimplechat.com/blog/react-suspense/ (using a hook) is better than the current one in this PR. I am fine with keeping Suspense but we should follow the hook usage instead of passing a promise as a prop (that looks like a code smell in React). As the same time, if you prefer to get rid of the whole Suspense bit, I am good with it too. The choice is yours.

Reply:
Actually I tried that way first!

It worked, but it is only applicable if we use that hook in the child component(`ListNewsletter`). I need to use in the parent component(`Home`), because I will need to pass the `getData` function to other components on the incoming PRs to re-render the newsletter list after the user update or delete a newsletter.

Seems like we should go back to the traditional way 👀. I also find `Suspense` seems more suitable for cases when we want to load only an independent component, like, ViewNewsletterDetail, instead of a big list of Newsletters. 😄

Reply:
The `Suspense` is removed! Resolved [here](https://github.com/nimblehq/js-framework-benchmark/pull/110/commits/da187feeeca914879b6d502720a739b9205f3408#diff-223e9b714cf2c9b5f163079f0bd63d3a6eac9c46915fc3afde20a2742897d706R54)